slasher/db/kv/spanner.go

```diff
@@ -370,13 +375,19 @@ func (db *Store) findOrLoadEpochInCache(ctx context.Context, epoch uint64) (map[
 	return spanForEpoch, nil
 }
 
-func setObservedEpochs(epoch uint64) {
+func (db *Store) setObservedEpochs(ctx context.Context, epoch uint64) error {
+	var err error
 	if epoch > highestObservedEpoch {
 		slasherHighestObservedEpoch.Set(float64(epoch))
 		highestObservedEpoch = epoch
+		// Prune block header history every PruneSlasherStoragePeriod epoch.
+		if highestObservedEpoch%params.BeaconConfig().PruneSlasherStoragePeriod == 0 {
+			err = db.PruneAttHistory(ctx, epoch, params.BeaconConfig().WeakSubjectivityPeriod)
```

Comment:
Any reason not to handle the error immediately here?

Reply:
changed